src/Microsoft.ML.FastTree/Training/Parallel/SingleTrainer.cs

```diff
@@ -19,37 +19,37 @@ namespace Microsoft.ML.Trainers.FastTree
     using LeafSplitCandidates = Internal.LeastSquaresRegressionTreeLearner.LeafSplitCandidates;
     using SplitInfo = Internal.LeastSquaresRegressionTreeLearner.SplitInfo;
 
-    public sealed class SingleTrainer : IParallelTraining
+    internal sealed class SingleTrainer : IParallelTraining
     {
-        public void CacheHistogram(bool isSmallerLeaf, int featureIdx, int subfeature, SufficientStatsBase sufficientStatsBase, bool hasWeights)
+        void IParallelTraining.CacheHistogram(bool isSmallerLeaf, int featureIdx, int subfeature, SufficientStatsBase sufficientStatsBase, bool hasWeights)
```

Comment:
private void

Reply:
No. This is an explicit implementation of an internal interface.

---
In reply to: [251976714](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251976714) [](ancestors = 251976714)